src/docker/Dockerfile

```diff
@@ -56,39 +57,24 @@ RUN apt-get install -y software-properties-common python-software-properties
 RUN add-apt-repository ppa:git-core/ppa
 RUN apt-get update && apt-get install -y git git-core
 
-# SSH Daemon
-RUN apt-get install -y ssh && \
-  mkdir /var/run/sshd && \
-  chmod 0755 /var/run/sshd
+# Java
+RUN apt-get update && apt-get install -y \
+    openjdk-7-jdk="${OPENJDK_VERSION}"
 
-# Ruby
-RUN apt-get install -y libmagic-dev zlib1g-dev
+# Ruby; for puzzle creation
+RUN apt-get update && apt-get install -y \
+    libmagic-dev zlib1g-dev
 RUN apt-add-repository ppa:brightbox/ruby-ng
-RUN apt-get update
-RUN apt-get install -y ruby2.2 ruby2.2-dev
-RUN gem update
-RUN gem install nokogiri
-RUN gem install bundler
```

Comment:
@amihaiemil I think that `nokogiri` and `bundler` may also be required for puzzle creation